pandas/io/stata.py

```diff
@@ -1518,11 +1518,8 @@ def _read_strls(self) -> None:
             self.GSO[str(v_o)] = decoded_va
 
     def __next__(self) -> DataFrame:
-        if self._chunksize is None:
-            raise ValueError(
-                "chunksize must be set to a positive integer to use as an iterator."
-            )
-        return self.read(nrows=self._chunksize or 1)
+        self._chunksize = 1 if self._chunksize is None else self._chunksize
```

Comment:
is there validation when chunsize is passed in?

Reply:
It is not checked when set. I've refactored the code to not rely on chunksize being set to determine if an iterator is being used.

Reply:
what i mean is what if someone passes `chunksize<=0` ? do we appropriately raise

Reply:
Yes, there is explicit  validation to ensure that it is an `int` that is `>0`

https://github.com/pandas-dev/pandas/blob/b2ea05f46fa9d65d8176ea0217276b7c56549924/pandas/io/stata.py#L1049-L1052